base/iobuffer.jl

```diff
@@ -92,6 +92,13 @@ end
     return byte
 end
 
+@inline function unsafe_read(from::AbstractIOBuffer, ::Type{UInt8})
```

Comment:
The naming concern is fair. I didn't put much thought into it. Using boundscheck would be interesting, so you could then do
```julia 
@inbounds b = read(io, UInt8)
```
?